src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java

```diff
@@ -116,12 +118,31 @@ public AWSCredentials getCredentials() {
         if (StringUtils.isBlank(iamRoleArn)) {
             return initialCredentials;
         } else {
+            // Check for available region from the SDK, otherwise specify default
+            String clientRegion;
+            DefaultAwsRegionProviderChain sdkRegionLookup = new DefaultAwsRegionProviderChain();
+            try {
+                clientRegion = sdkRegionLookup.getRegion();
+            }
+            catch(com.amazonaws.SdkClientException e) {
+                if (e.getMessage() == "Unable to load region information from any provider in the chain") {
```

Comment:
Matching in the exception message is pretty weak. In this case I would log the exception and move forward and outside the `try` block if `clientRegion` is still `null`, set it to `Regions.DEFAULT_REGION.getName()`